### PR TITLE
bugfix(Whitebox): better handle updating material for whitebox mesh (GHI-9442)

### PR DESCRIPTION
**Pull request by pollend**
looks like ApplyProperties is failing so the mesh will disappear. this is a continuation of the previous fix. 

https://user-images.githubusercontent.com/854359/169208246-afa59133-2f41-41a7-b16c-2c6d7326cc42.mp4

ref: https://github.com/o3de/o3de/pull/9442#event-6617921824

### CODE REVIEW

File: Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp [hunk 1]
```diff
@@ -261,13 +262,37 @@ namespace WhiteBox
         m_meshFeatureProcessor->SetTransform(m_meshHandle, worldFromLocal);
     }
 
+
+
```
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp:266]: uber nit: Can we please remove these line breaks

File: Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.h [hunk 2]
```diff
@@ -49,6 +51,9 @@ namespace WhiteBox
         bool IsVisible() const override;
         void SetVisiblity(bool visibility) override;
 
+        // AZ::TickBus overrides...
```
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.h:54]: uber nit: space after overrides (`AZ::TickBus overrides ...`)

File: Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp [hunk 3]
```diff
@@ -261,13 +262,37 @@ namespace WhiteBox
...
+                m_meshFeatureProcessor->SetMaterialAssignmentMap(m_meshHandle, m_materialMap);
+            }
+            else if (!AZ::TickBus::Handler::BusIsConnected())
+            {
+                AZ::TickBus::Handler::BusConnect();
+            }
+        }
+    }
+
+    void AtomRenderMesh::OnTick([[maybe_unused]]float deltaTime, [[maybe_unused]]AZ::ScriptTimePoint time) {
```
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp:289]: nit/standards: Opening brace should be on a new line for the function

Comment (hultonha) [reply Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp:289]: (maybe try clang-format for these added lines?)

File: Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp [hunk 4]
```diff
@@ -261,13 +262,37 @@ namespace WhiteBox
...
+
+
     void AtomRenderMesh::UpdateMaterial(const WhiteBoxMaterial& material)
     {
+        if (m_meshFeatureProcessor)
+        {
+            auto& materialAssignment = m_materialMap[AZ::Render::DefaultMaterialAssignmentId];
+            materialAssignment.m_propertyOverrides[AZ::Name("baseColor.color")] = AZ::Color(material.m_tint);
+            materialAssignment.m_propertyOverrides[AZ::Name("baseColor.useTexture")] = material.m_useTexture;
+            if (materialAssignment.ApplyProperties())
```
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp:274]: suggestion: A guide comment here might be useful to let the reader know what is happening, something like... `if ApplyProperties fails, defer updating the material assignment map to the next tick, and try applying properties again`

**Review by hultonha — APPROVED** [review]
Change looks okay to me, would be good to have @amzn-tommy weigh in just to sanity check from the Atom side. Be good for @jonawals to take a look too.

**Review by jonawals — APPROVED** [review]
Nothing more to add that hasn't already been said by other reviewers. Great to see WhiteBox shaping up with proper support for materials!

**Review by amzn-tommy — APPROVED** [review]

**Comment by hultonha** [comment]
Just kicked off a build now, thanks @pollend!

**Review by cgalvan — APPROVED** [review]

**Pull request merged**